kmir/k-src/mir.md

```diff
@@ -50,9 +50,19 @@ If we are, then we stop execution and enter the finalization phase. Otherwise, i
   rule <k> #return(FUNCTION_KEY, _) => .K ... </k>
        <callStack> ListItem(FUNCTION_KEY) XS => XS </callStack>
     requires FUNCTION_KEY =/=K Fn(main)
+```
+
+The `#halt` construct is used to signify the end of execution. Any remaining items on the `<k>` cell will be removed.
+
+```k
+  syntax KItem ::= "#halt"
+  //----------------------
+  rule [halt]: <k> #halt ~> (_:MirSimulation => .K) ... </k>
+  rule         <k> #halt ~> (#initialized()  => .K) ... </k>
```

Comment:
Is the intent of these rules to throw away the entire contents of the `<k>` cell following the `#halt`? If so you probably want to rewrite them without structural framing (`...`); I'm not 100% convinced they'll do what you expect here (but I might be wrong!).

Reply:
I actually had the same thought! I used EVM as a model, and saw [they used structural framing for this](https://github.com/runtimeverification/evm-semantics/blob/master/kevm-pyk/src/kevm_pyk/kproj/evm-semantics/evm.md#exception-based). I figured that it must be working then, and I wasn't sure of the nuance that might be behind leaving it. I tested and it is working without it as far as I can tell, I think I will leave it off.